### PR TITLE
bump:  1.20＋updates

### PR DESCRIPTION
**Pull request by Wudji**
- Note: I created a fork which merged changes in several PRs: https://github.com/Wudji/autofish
- (You can get the mod file at https://modrinth.com/mod/x+-autofish)

### CODE REVIEW

**Comment by FudgeDumbo** [comment]
it compiles and loads without an issue but when fish bites it does not reel/pull

File: src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class FishMonitorMPMotion implements FishMonitorMP {
 
     @Override
     public void hookTick(Autofish autofish, MinecraftClient minecraft, FishingBobberEntity hook) {
-        if (worldContainsBlockWithMaterial(hook.world, hook.getBoundingBox(), Material.WATER)) {
+        if (worldContainsBlockWithMaterial(hook.getWorld(), hook.getBoundingBox())) {
```
Comment (airwsw) [inline src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java]: Should be ported to latest APIs, not just removed

```diff
diff --git a/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java b/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
index e22cd73..55e659e 100644
--- a/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
+++ b/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
@@ -1,11 +1,11 @@
 package troy.autofish.monitor;
 
-import net.minecraft.block.Material;
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.entity.projectile.FishingBobberEntity;
-import net.minecraft.network.Packet;
+import net.minecraft.network.packet.Packet;
 import net.minecraft.network.packet.s2c.play.EntityVelocityUpdateS2CPacket;
-import net.minecraft.util.function.MaterialPredicate;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.MathHelper;
@@ -30,7 +30,7 @@ public class FishMonitorMPMotion implements FishMonitorMP {
 
     @Override
     public void hookTick(Autofish autofish, MinecraftClient minecraft, FishingBobberEntity hook) {
-        if (worldContainsBlockWithMaterial(hook.world, hook.getBoundingBox(), Material.WATER)) {
+        if (worldContainsBlockWithMaterial(hook.getWorld(), hook.getBoundingBox(), Blocks.WATER)) {
             hasHitWater = true;
         }
     }
@@ -71,16 +71,15 @@ public class FishMonitorMPMotion implements FishMonitorMP {
         }
     }
 
-    public static boolean worldContainsBlockWithMaterial(World world, Box box, Material material) {
+    public static boolean worldContainsBlockWithMaterial(World world, Box box, Block block) {
         int i = MathHelper.floor(box.minX);
         int j = MathHelper.ceil(box.maxX);
         int k = MathHelper.floor(box.minY);
         int l = MathHelper.ceil(box.maxY);
         int m = MathHelper.floor(box.minZ);
         int n = MathHelper.ceil(box.maxZ);
-        MaterialPredicate materialPredicate = MaterialPredicate.create(material);
         return BlockPos.stream(i, k, m, j - 1, l - 1, n - 1).anyMatch((blockPos) -> {
-            return materialPredicate.test(world.getBlockState(blockPos));
+            return world.getBlockState(blockPos).getBlock() != block;
         });
     }
 }
```

File: src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java [hunk 2]
```diff
@@ -71,16 +71,18 @@ public void handlePacket(Autofish autofish, Packet<?> packet, MinecraftClient mi
         }
     }
 
-    public static boolean worldContainsBlockWithMaterial(World world, Box box, Material material) {
+    public static boolean worldContainsBlockWithMaterial(World world, Box box/*, Material material*/) {
```
Comment (airwsw) [inline src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java]: same here

**Comment by Wudji** [comment]
[bandicam 2023-06-10 20-25-16-898.zip](https://github.com/MrTroot/autofish/files/11712420/bandicam.2023-06-10.20-25-16-898.zip)
it works for me :|

**Comment by airwsw** [comment]
Have you enabled the "Use Sound Detection"?

**Comment by Wudji** [comment]
> Have you enabled the "Use Sound Detection"?

No, and Sound Detection is disabled in singleplayer.
I haven't test the mod in multiplayer, maybe i should give it a test(

File: src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java [hunk 3]
```diff
@@ -71,16 +71,18 @@ public void handlePacket(Autofish autofish, Packet<?> packet, MinecraftClient mi
...
         int k = MathHelper.floor(box.minY);
         int l = MathHelper.ceil(box.maxY);
         int m = MathHelper.floor(box.minZ);
         int n = MathHelper.ceil(box.maxZ);
-        MaterialPredicate materialPredicate = MaterialPredicate.create(material);
+        // MaterialPredicate materialPredicate = MaterialPredicate.create(material);
         return BlockPos.stream(i, k, m, j - 1, l - 1, n - 1).anyMatch((blockPos) -> {
-            return materialPredicate.test(world.getBlockState(blockPos));
+            // world.getBlockState(blockPos).contains(new Property<T>() {});
+            return world.getBlockState(blockPos).contains(Properties.LEVEL_8);
```
Comment (airwsw) [inline src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java]: Can you explain, why a block state containing the `Properties.LEVEL_8` is a `WATER` block? Or guarantee that they work the exactly same.

Comment (Wudji) [reply src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java]: `LEVEL_8 = IntProperty.of("level", 0, 8);` in `net.minecraft.state.property.Properties`

and [it says](https://minecraft.fandom.com/wiki/Block_states#Water) block states "0" stands for the source of water(the highest fluid level).

But it somehow doesn't work in multiplayer(

**Comment by FudgeDumbo** [comment]
confirmed the new change worked :) thank you!

**Comment by 52nomis9** [comment]
How can I download and use it?Thank u

**Comment by Wudji** [comment]
> How can I download and use it?Thank u

you can get the jar file at https://github.com/Wudji/autofish/actions/runs/5229994451

**Comment by ameow1004** [comment]
I can't find the jar file you mentioned in the link, can you tell me where it is?

**Comment by Drift91** [comment]
@ameow1004 Click on the word "Artifact" with the cube icon to the left of it, halfway down the page.

**Comment by nikissXI** [comment]
nice job, bro! it works perfect in 1.20.1

**Comment by Frontesque** [comment]
> > How can I download and use it?Thank u
> 
> you can get the jar file at https://github.com/Wudji/autofish/actions/runs/5229994451

Since GitHub only hosts artifacts for 1 month after their creation, I've created a mirror on my CDN. I've tested the jar on 1.20.1 and its working fine. If the author wants me to take this down, please let me know. I'm just providing this to make it easier for other people to use. 

MIRROR: [https://emerald.frontesque.me/builds/autofish-0.9.10-SNAPSHOT_(Wudji).jar](https://emerald.frontesque.me/builds/autofish-0.9.10-SNAPSHOT_(Wudji).jar)
image

edit: grammar
edit: Link moved: [https://builds.frontesque.me/download.php?id=2&token=hq472krdvp2Tv99MovOpvMkoxqG2hKk5](https://builds.frontesque.me/download.php?id=2&token=hq472krdvp2Tv99MovOpvMkoxqG2hKk5)

**Comment by James-P-Bennett** [comment]
Excellent work, everyone! Appreciate the 1.20 update. :)

**Review by James-P-Bennett — APPROVED** [review]
Works great on 1.20.2

**Comment by Frontesque** [comment]
@Wudji Thanks again for another awesome update. The mirror can be found below:
[1.20.4 Build Download](https://builds.frontesque.me/download.php?id=5&token=SL0YEW7DaKb1IU0qKNO3ydcbkfbUWV5a)

**Pull request closed without merging**